### ISSUE TITLE
Globalization rules should not be raised when InvariantGlobalization is set

### ISSUE THREAD

**Issue by ackginger** [dotnet/roslyn-analyzers#4629]
### Analyzer

https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/globalization-warnings

### Describe the improvement

Currently these rules are all firing (e.g. CA1310 Specify StringComparison for correctness) despite having InvariantGlobalization turned on in the project file

```
<PropertyGroup>
    <InvariantGlobalization>true</InvariantGlobalization>
</PropertyGroup>
```

as per https://github.com/dotnet/runtime/blob/master/docs/design/features/globalization-invariant-mode.md

This option makes the warnings moot, so they should not be raised.

**Comment by Evangelink** [comment]
Interesting, I wasn't aware of this option. I think this suggestion makes sense.

@mavasani there is a little helper (I think you wrote it) to read project option, do you know if it also reads the json and environment variables?

**Comment by Youssef1313** [comment]
@mavasani @Evangelink Do we need to introduce a base abstract diagnostic analyzer class for globalization rules to ensure the check is done for all globalization rules? and potentially introduce an analyzer that's only used internally to make sure any globalization rule derives from this class?

**Comment by Evangelink** [comment]
I think that would be easier than ensuring all globalization rules call the helper.

**Comment by Youssef1313** [comment]
@Evangelink I'm now facing a problem where an analyzer have two rules, only one of which has Globalization category:

https://github.com/dotnet/roslyn-analyzers/blob/f167914d442d0ae0055b28a1c92e5b9d446fdbdb/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PInvokeDiagnosticAnalyzer.cs

If I made the analyzer inherits `AbstractGlobalizationDiagnosticAnalyzer`, it will bail-out for both rules (not only the globalization one). Do you have a good idea to fix that?

**Comment by Evangelink** [comment]
Hum I wasn't expecting that... 

I'd say we can do:
- your suggestion
- both a base class and a helper, so that caller decides what to use
- revert this base class and only use a helper method
- make a post-filtering (check at the time where we report the diagnostic)
